aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -269,6 +269,62 @@ void printTensorNDArray(const Tensor& t) {
   return cachedGraph;
 }
 
+id<MTLBuffer> scatterViewTensor(at::Tensor& dst, const at::Tensor src, id<MTLBuffer> updatesTensorBuffer) {
+  MPSCachedGraph* mpsCachedGraph = _getCachedGraph(dst);
+
+  TORCH_CHECK(mpsCachedGraph != nil);
+
+  MPSStream* stream = getCurrentMPSStream();
+
+  struct CachedGraph : public MPSCachedGraph
+  {
+    CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+    MPSGraphTensor* inputTensor_ = nil;
+    MPSGraphTensor* outputTensor_ = nil;
+
+    MPSGraphTensor* scatteredTensor_ = nil;
+    MPSGraphTensor* updatesTensor_ = nil;
+  };
+
+  CachedGraph* cachedGraph = static_cast<CachedGraph *>(mpsCachedGraph);
+  TORCH_CHECK(cachedGraph != nil);
+  TORCH_CHECK(cachedGraph->scatteredTensor_ != nil);
+  TORCH_CHECK(cachedGraph->updatesTensor_ != nil);
+
+  @autoreleasepool {
+    MPSGraphTensor* inputTensor = cachedGraph->inputTensor_;
+    MPSShape *shape = [inputTensor shape];
+    auto dataType = [inputTensor dataType];
+    // In-place scatter operation
+    id<MTLBuffer> dataBuffer = __builtin_bit_cast(id<MTLBuffer>, dst.storage().data());
+
+    MPSGraphTensorData* inputTensorData = [[[MPSGraphTensorData alloc] initWithMTLBuffer: dataBuffer
```

Comment:
add `autorelease` here.

Reply:
Same for all the alloc below.

Reply:
this contains the autorelease couple of lines below (line [303](https://github.com/kulinseth/pytorch/pull/38/files#diff-c35f3c3f2f56d77db0e1b290ea4ca49e78e016c5732d319863ebb1c88cd775bcR303) for this one)